### ISSUE TITLE
Add support for ES modules

### ISSUE THREAD

**Issue by schoero** [benyap/resolve-tspaths#138]
Hi

First of all, thank you for your efforts. Working with tsconfig paths is very frustrating and your module is the closest one to work for me, out of 5 different ones I had to try.

But in order to compile to ES modules, we [have to use relative paths, including a `.js` extension](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#type-in-package-json-and-new-extensions).

### Example

Imagine I have two files:

```txt
src/
├ compiler/
│   └ index.ts
└ bin/
    └ index.ts
```

And a `tsconfig.json` like the following:

```json
{
  "compilerOptions": {
    "module": "ESNext",
    "baseUrl": ".",
    "paths": {
      "compiler/*": ["src/compiler/*"],
    }
  }
}
```
If I have an import like this
```ts
// src/bin/index.ts
import { compile } from "compiler/index.js"; // actual file is src/compiler/index.ts
```

It should resolve to this:
```ts
// src/bin/index.ts
import { compile } from "../compiler/index.js";
```

Basically we have to import the file with a `.js` extension, even though the actual file has a `.ts` extension.

This currently does not work because of the `existsSync()` check in `generateChanges.ts`, which adds `ext` to the `modulePath` that already has the `.js` extension from the import.

https://github.com/benyap/resolve-tspaths/blob/5754ef5052477bfa4a9821b19302dfa6f998a783/src/steps/generateChanges.ts#L143

Furthermore the file extension currently gets removed in the replaced path but the file extension should stay as it is in the original import.

**Comment by benyap** [comment]
Hey @schoero, thanks for raising this issue. Appreciate the detail you've provided too - very helpful especially if it's something I'm not familiar with! Will look into this as soon as I can when I have time.

In the mean time, if you have any more examples, test cases or edge cases you can think of, please let me know as they'll be invaluable when I get around to implementation and testing.

**Comment by schoero** [comment]
Hi @benyap, thank you for your response. No pressure from my end. I got it working for my use case with some [quick and dirty changes](https://github.com/schoero/resolve-tspaths/commit/3278c592776d5484f4f6935c2dc3e8594efe4ebb). They also pass the existing tests :)

I currently only have one project that outputs ES modules and uses the "paths" feature of TypeScript and I didn't had too much time to think about this in the last few days, so I can't provide much more information right now. 

But I will get back to this as soon as possible.

**Comment by Zebreus** [comment]
TypeScript imports are also based on how nodejs [resolve commonjs modules](https://nodejs.org/api/modules.html#all-together), but nodejs uses a [slightly different strategy](https://nodejs.org/api/esm.html#terminology) for resolving ES modules. The biggest change is that extensions are mandatory for relative imports. I think fixing missing extensions also belongs into this package.

I think it would make sense to fix the extensions, if TypeScript always generates ES modules.

**Comment by benyap** [comment]
@schoero support for ES modules has been released in [v0.8.0](https://github.com/benyap/resolve-tspaths/releases/tag/v0.8.0). Feel free to give it a shot and leave any feedback you have!
